inc/namespace.php

```diff
@@ -264,7 +282,21 @@ function is_amf_asset( WP_Post $attachment ) : bool {
 	return strpos( $attachment->post_name, 'amf-' ) === 0;
 }
 
-function fix_duplicate_baseurl( string $url, WP_Post $attachment ) : string {
+function get_asset_provider( ?WP_Post $attachment ) : ?Provider {
+	if ( empty( $attachment ) || ! is_amf_asset( $attachment ) ) {
+		return null;
+	}
+
+	try {
+		$provider_id = wp_unslash( get_post_meta( $attachment->ID, '_amf_provider', true ) );
+		return ProviderRegistry::instance()->get( $provider_id );
+	} catch ( Exception $e ) {
+		trigger_error( $e->getMessage(), E_USER_WARNING );
+		return null;
+	}
```

Comment:
To have a single `return` (after the early bail), one **could** write this like so:

```suggestion
	$provider = null;

	try {
		$provider_id = wp_unslash( get_post_meta( $attachment->ID, '_amf_provider', true ) );
		$provider = ProviderRegistry::instance()->get( $provider_id );
	} catch ( Exception $e ) {
		trigger_error( $e->getMessage(), E_USER_WARNING );
	}

	return $provider;
```

This wouldn't have `return` expression in both `try` and `catch`, too.